src/DataTypes/convertMySQLDataType.cpp

```diff
@@ -25,52 +30,88 @@ ASTPtr dataTypeConvertToQuery(const DataTypePtr & data_type)
     return makeASTFunction("Nullable", dataTypeConvertToQuery(typeid_cast<const DataTypeNullable *>(data_type.get())->getNestedType()));
 }
 
-DataTypePtr convertMySQLDataType(const std::string & mysql_data_type, bool is_nullable, bool is_unsigned, size_t length)
+DataTypePtr convertMySQLDataType(MultiEnum<MySQLDataTypesSupport> type_support,
```

Comment:
I'm not sure. I think we should remove it(but maybe not now).

FYI:
https://github.com/ClickHouse/ClickHouse/pull/11214
https://github.com/ClickHouse/ClickHouse/pull/13761
https://github.com/ClickHouse/ClickHouse/pull/10851#discussion_r440377052

Reply:
Hi, could you please elaborate? Remove what exactly?

Reply:
Most MySQL types have aliases in ClickHouse. Perhaps we should use the `DataTypeFactory` to get these(directly use mysql types name) types.

Reply:
@zhang2014 Am I right that you suggest to implement `DataTypeFactory::getFromMySQLType()` and move al this code there?

Reply:
Just suggest(may not be appropriate to modify now):

We can simply concat a string to create a type description and use it as a argument to call `DataTypeFactory::get()`. because we already have aliases of some types for better SQL compatibility.

BTW: the MaterializeMySQL engine is working in this manner. But its compatibility and stability are uncertain(I only confirm that it satisfies most types in my env)

Reply:
1. I don't really like the idea of concatenating stuff into string just to squeeze it throught interface of `DataTypeFactory::get()`